components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQTask.java

```diff
@@ -23,27 +23,38 @@
 import org.apache.synapse.core.SynapseEnvironment;
 import org.wso2.carbon.inbound.endpoint.common.OneTimeTriggerInboundTask;
 
+/**
+ * The task implementation to trigger the RabbitMQ consumer.
+ */
 public class RabbitMQTask extends OneTimeTriggerInboundTask {
 
     private static final Log log = LogFactory.getLog(RabbitMQTask.class.getName());
 
-    private RabbitMQConnectionConsumer rabbitMQConnectionConsumer;
+    private RabbitMQConsumer rabbitMQConsumer;
 
-    public RabbitMQTask(RabbitMQConnectionConsumer rabbitMQConnectionConsumer) {
-        log.debug("RabbitMQ Task initialize.");
-        this.rabbitMQConnectionConsumer = rabbitMQConnectionConsumer;
+    public RabbitMQTask(RabbitMQConsumer rabbitMQConsumer) {
+        if (log.isDebugEnabled()) {
+            log.debug("RabbitMQ Task initialize.");
+        }
+        this.rabbitMQConsumer = rabbitMQConsumer;
     }
 
     protected void taskExecute() {
-        log.debug("Executing RabbitMQ Task Execution.");
-        rabbitMQConnectionConsumer.execute();
+        if (log.isDebugEnabled()) {
```

Comment:
We do not need to check if debug is enabled since there is no string concatenation. The check will be done internally.

Reply:
Ack.